[2025-04-23T18:45:06.783+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: mbta_kafka_consumer_dag.run_kafka_consumer scheduled__2025-04-23T18:20:00+00:00 [queued]>
[2025-04-23T18:45:06.892+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: mbta_kafka_consumer_dag.run_kafka_consumer scheduled__2025-04-23T18:20:00+00:00 [queued]>
[2025-04-23T18:45:06.893+0000] {taskinstance.py:2170} INFO - Starting attempt 2 of 2
[2025-04-23T18:45:07.012+0000] {taskinstance.py:2191} INFO - Executing <Task(PythonOperator): run_kafka_consumer> on 2025-04-23 18:20:00+00:00
[2025-04-23T18:45:07.152+0000] {standard_task_runner.py:60} INFO - Started process 219 to run task
[2025-04-23T18:45:07.208+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'mbta_kafka_consumer_dag', 'run_kafka_consumer', 'scheduled__2025-04-23T18:20:00+00:00', '--job-id', '510', '--raw', '--subdir', 'DAGS_FOLDER/mbta_kafka_consumer_dag.py', '--cfg-path', '/tmp/tmp6tx_khhh']
[2025-04-23T18:45:07.485+0000] {standard_task_runner.py:88} INFO - Job 510: Subtask run_kafka_consumer
[2025-04-23T18:45:09.763+0000] {task_command.py:423} INFO - Running <TaskInstance: mbta_kafka_consumer_dag.run_kafka_consumer scheduled__2025-04-23T18:20:00+00:00 [running]> on host f5dfe1f680f3
[2025-04-23T18:45:11.201+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='mbta_kafka_consumer_dag' AIRFLOW_CTX_TASK_ID='run_kafka_consumer' AIRFLOW_CTX_EXECUTION_DATE='2025-04-23T18:20:00+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-04-23T18:20:00+00:00'
[2025-04-23T18:45:11.222+0000] {connection.py:370} INFO - Snowflake Connector for Python Version: 3.6.0, Python Version: 3.9.18, Platform: Linux-6.10.14-linuxkit-aarch64-with-glibc2.36
[2025-04-23T18:45:11.307+0000] {connection.py:1171} INFO - This connection is in OCSP Fail Open Mode. TLS Certificates would be checked for validity and revocation status. Any other Certificate Revocation related exceptions or OCSP Responder failures would be disregarded in favor of connectivity.
[2025-04-23T18:45:14.242+0000] {mbta_kafka_consumer.py:45} INFO - ‚úÖ Connected to Snowflake successfully.
[2025-04-23T18:45:14.271+0000] {mbta_kafka_consumer.py:220} INFO - üöÄ Starting MBTA Kafka Consumer...
[2025-04-23T18:45:14.409+0000] {conn.py:395} INFO - <BrokerConnection client_id=kafka-python-2.1.5, node_id=bootstrap-0 host=kafka:9092 <connecting> [IPv4 ('172.19.0.4', 9092)]>: connecting to kafka:9092 [('172.19.0.4', 9092) IPv4]
[2025-04-23T18:45:14.470+0000] {conn.py:615} INFO - <BrokerConnection client_id=kafka-python-2.1.5, node_id=bootstrap-0 host=kafka:9092 <checking_api_versions_recv> [IPv4 ('172.19.0.4', 9092)]>: Broker version identified as 2.6
[2025-04-23T18:45:14.474+0000] {conn.py:456} INFO - <BrokerConnection client_id=kafka-python-2.1.5, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.19.0.4', 9092)]>: Connection complete.
[2025-04-23T18:45:14.513+0000] {consumer.py:119} WARNING - group_id is None: disabling auto-commit.
[2025-04-23T18:45:14.529+0000] {subscription_state.py:182} INFO - Updating subscribed topics to: ('mbta_alerts', 'mbta_routes', 'mbta_stops', 'mbta_vehicles', 'mbta_predictions')
[2025-04-23T18:45:14.605+0000] {subscription_state.py:254} INFO - Updated partition assignment: [('mbta_stops', 0), ('mbta_predictions', 0), ('mbta_routes', 0), ('mbta_vehicles', 0), ('mbta_alerts', 0)]
[2025-04-23T18:45:14.728+0000] {conn.py:395} INFO - <BrokerConnection client_id=kafka-python-2.1.5, node_id=1 host=kafka:9092 <connecting> [IPv4 ('172.19.0.4', 9092)]>: connecting to kafka:9092 [('172.19.0.4', 9092) IPv4]
[2025-04-23T18:45:14.745+0000] {conn.py:456} INFO - <BrokerConnection client_id=kafka-python-2.1.5, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.19.0.4', 9092)]>: Connection complete.
[2025-04-23T18:45:14.760+0000] {conn.py:936} INFO - <BrokerConnection client_id=kafka-python-2.1.5, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.19.0.4', 9092)]>: Closing connection. 
[2025-04-23T18:45:43.381+0000] {mbta_kafka_consumer.py:234} INFO - üì• Received 1000 records from topic: mbta_stops
[2025-04-23T18:45:44.624+0000] {mbta_kafka_consumer.py:56} INFO - üîç Sample data for MBTA_STOPS:
[2025-04-23T18:45:44.689+0000] {mbta_kafka_consumer.py:57} INFO - 
          STOP_ID  ...                 CREATED_AT
0            5682  ... 2025-04-23 18:45:43.450986
1           50620  ... 2025-04-23 18:45:43.451037
2  node-417-lobby  ... 2025-04-23 18:45:43.451040
3            4042  ... 2025-04-23 18:45:43.451052
4          109822  ... 2025-04-23 18:45:43.451084

[5 rows x 5 columns]
[2025-04-23T18:45:45.600+0000] {cursor.py:1028} INFO - query: [SELECT * FROM MBTA_STOPS WHERE STOP_ID IN ('5682','50620','node-417-lobby','4042...]
[2025-04-23T18:45:46.264+0000] {cursor.py:1041} INFO - query execution done
[2025-04-23T18:45:46.440+0000] {cursor.py:1205} INFO - Number of results in first chunk: 0
[2025-04-23T18:45:53.860+0000] {mbta_kafka_consumer.py:104} INFO - üîÅ 0 new/updated records out of 1000 incoming records (some may be duplicate VEHICLE_IDs) for MBTA_STOPS
[2025-04-23T18:45:53.872+0000] {mbta_kafka_consumer.py:107} INFO - ‚úÖ All 1000 records already up-to-date in MBTA_STOPS, no merge needed.
[2025-04-23T18:45:53.877+0000] {mbta_kafka_consumer.py:234} INFO - üì• Received 10 records from topic: mbta_predictions
[2025-04-23T18:45:53.929+0000] {mbta_kafka_consumer.py:56} INFO - üîç Sample data for MBTA_PREDICTIONS:
[2025-04-23T18:45:53.936+0000] {mbta_kafka_consumer.py:57} INFO - 
                               PREDICTION_ID  ...                 CREATED_AT
0           prediction-67995089-70079-90-Red  ... 2025-04-23 18:45:53.881737
1  prediction-ADDED-1582321358-70080-130-Red  ... 2025-04-23 18:45:53.881791
2  prediction-ADDED-1582321378-70080-130-Red  ... 2025-04-23 18:45:53.881817
3           prediction-67995093-70079-90-Red  ... 2025-04-23 18:45:53.881845
4          prediction-67995120-70080-130-Red  ... 2025-04-23 18:45:53.881850

[5 rows x 9 columns]
[2025-04-23T18:45:54.198+0000] {cursor.py:1028} INFO - query: [SELECT * FROM MBTA_PREDICTIONS WHERE PREDICTION_ID IN ('prediction-67995089-7007...]
[2025-04-23T18:45:54.517+0000] {cursor.py:1041} INFO - query execution done
[2025-04-23T18:45:54.527+0000] {cursor.py:1205} INFO - Number of results in first chunk: 10
[2025-04-23T18:45:54.715+0000] {mbta_kafka_consumer.py:104} INFO - üîÅ 10 new/updated records out of 10 incoming records (some may be duplicate VEHICLE_IDs) for MBTA_PREDICTIONS
[2025-04-23T18:45:54.726+0000] {cursor.py:1028} INFO - query: [MERGE INTO MBTA_PREDICTIONS AS target USING (SELECT 'prediction-67995089-70079-9...]
[2025-04-23T18:45:56.711+0000] {cursor.py:1041} INFO - query execution done
[2025-04-23T18:45:56.724+0000] {cursor.py:1028} INFO - query: [MERGE INTO MBTA_PREDICTIONS AS target USING (SELECT 'prediction-ADDED-1582321358...]
[2025-04-23T18:45:58.388+0000] {local_task_job_runner.py:234} INFO - Task exited with return code -9
[2025-04-23T18:45:59.219+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
